PhotoEditor/app/src/androidTest/java/com/microsoft/device/display/samples/photoeditor/PhotoEditorUITest.kt

```diff
@@ -0,0 +1,214 @@
+
+        // Unlock rotation
+        device.unfreezeRotation()
+    }
+
+    /**
+     * Tests drag and drop capabilities of PhotoEditor
+     *
+     * @precondition device in portrait mode, no other applications are open
+     * (so app by default opens on left screen)
+     */
+    @Test
+    fun testDragAndDrop() {
+        // Lock in portrait mode
+        device.setOrientationNatural()
+
+        // Rotate and save image file
+        onView(withId(R.id.rotate_left)).perform(click())
+        onView(withId(R.id.save)).perform(click())
+
+        // Reset to unrotated image
+        onView(withId(R.id.rotate_right)).perform(click())
+        val prev = activityRule.activity.findViewById<ImageFilterView>(R.id.image).drawable
+
+        val filesPackage =
+            if (Build.MODEL.contains("Emulator")) {
+                "com.android.documentsui" // emulator apk package name
+            } else {
+                "com.google.android.documentsui" // device apk package name
+            }
+
+        // Open Files app
+        val context = InstrumentationRegistry.getInstrumentation().context
+        val intent = context.packageManager.getLaunchIntentForPackage(filesPackage)?.apply {
+            addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK)
+        }
+        context.startActivity(intent)
+        device.wait(Until.hasObject(By.pkg(filesPackage).depth(0)), 3000) // timeout at 3 seconds
+
+        // Before import, drawable is equal to prev
+        assertThat(prev, iz(activityRule.activity.findViewById<ImageFilterView>(R.id.image).drawable))
+
+        // Hardcoded to select most recently saved file in Files app - must be an image file
+        device.swipe(1550, 1230, 1550, 1230, 100)
+
+        // Slowly drag selected image file to other screen for import
+        device.swipe(1550, 1230, 1200, 1100, 600)
+
+        // After import, drawable has changed
+        onIdle()
+        assertThat(
+            prev,
+            iz(not(activityRule.activity.findViewById<ImageFilterView>(R.id.image).drawable))
+        )
+
+        // Close Files app
+        closeRight()
+
+        // Unlock rotation
+        device.unfreezeRotation()
+    }
+
+    /**
+     * HELPER FUNCTIONS FOR DUAL-SCREEN BEHAVIOR
+     *
+     * Use the functions below in your tests when testing dual-screen behaviors and transitions.
+     * Run the "configureSpanning" test below to check that the methods produce the expected
+     * behavior on your device.
+     *
+     * If the test fails, modify the swipe parameters as needed- usually either an increase in the
+     * "steps" parameter or a slight shift in the "endX" parameter.
+     *
+     */
+
+    /**
+     * Runs helper functions and checks that they work as expected
+     *
+     * @precondition device in portrait mode, no other applications are open
+     * (so app by default opens on left screen)
+     */
+    @Test
+    fun configureSpanning() {
```

Comment:
Since this is a test, maybe a more test-specific name would be useful, something like:
`testSpanningScenarios` or something like that.
`condigureSpanning` sounds more like a precondition you require for your tests and not for a real test.

Reply:
good catch, just pushed up the change!